drivers/rilmodem/sim.c

```diff
@@ -943,7 +944,8 @@ static void ril_query_pin_retries(struct ofono_sim *sim,
 			g_free(cbd);
 			CALLBACK_WITH_FAILURE(cb, NULL, data);
 		}
-	} else if (sd->vendor == OFONO_RIL_VENDOR_MTK) {
+	} else if (sd->vendor == OFONO_RIL_VENDOR_MTK ||
+			sd->vendor == OFONO_RIL_VENDOR_MTK2) {
```

Comment:
Hmmm...  I don't really like this.  We're really quirking on a different modem design from the same vendor, not a new vendor.